CodeTree/2017-2018년/DH_토스트_계란틀.java

```diff
@@ -0,0 +1,99 @@
+import java.io.*;
+import java.util.*;
+
+/*
+토스트 계란틀
+ */
+
+public class DH_토스트_계란틀 {
+    static class Point {
+        int r, c;
+        public Point(int r, int c) {
+            this.r = r;
+            this.c = c;
+        }
+    }
+    static int N, L, R;
+    static int[][] map;
+    static boolean[][] v;
+    static int[] dr = {1, 0, -1, 0}, dc = {0, 1, 0, -1};
+    static ArrayDeque<Point> q, tmp;
+
+    static void solution() {
+        int turn = 0;
+        while(canMove()) turn++;
+        System.out.println(turn);
+    }
+
+    static boolean canMove() {
+        v = new boolean[N][N];
+        boolean flag = false;
+
+        for(int r = 0; r < N; r++) {
+
+            int sum, cnt;
+            for(int c = 0; c < N; c++) {
+                if(v[r][c]) continue;
+                sum = map[r][c];
+                cnt = 1;
+
+                q.add(new Point(r, c));
+                tmp.add(new Point(r, c));
+                v[r][c] = true;
+                while(!q.isEmpty()) {
+                    Point current = q.poll();
+                    tmp.add(current);
+
+                    for(int d = 0; d < 4; d++) {
+                        int nr = current.r + dr[d];
+                        int nc = current.c + dc[d];
+
+                        if(!check(nr, nc) || v[nr][nc] || !isInRange(map[current.r][current.c], map[nr][nc])) continue;
+                        v[nr][nc] = true;
+                        q.add(new Point(nr, nc));
+                        tmp.add(new Point(nr, nc));
+                        sum += map[nr][nc];
+                        cnt++;
+                        flag = true;
+                    }
+                }
+
+                while(!tmp.isEmpty()) {
+                    Point current = tmp.poll();
+                    map[current.r][current.c] = sum / cnt;
+                }
+            }
+
+        }
+        return flag;
+    }
+
+    static boolean isInRange(int a, int b) {
+        return Math.abs(a - b) >= L && Math.abs(a - b) <= R;
```

Comment:
`Math.abs(a - b)` 같은 결과값이 두번 작성되어서 변수로 따로 받아서 1번만 작성하는게 좋을 것 같습니다 ㅎㅎ
```suggestion
        int abs = Math.abs(a - b);
        return abs >= L && abs  <= R;
```